eclipse-cdt-cloud/vscode-memory-inspector
# Variables and Variable Decorations

#### What it does


- Fixes #12 - Adds plugin-side infrastructure for contributing non-DAP functionality (oriented toward variable retrieval, for now) and implements variable retrieval for GDB debugging.
- Closes #13 - Adds webview-side infrastructure for contributing CSS-based decorations to particular ranges of memory, and implements an example for variables
- Resolves #14 - Adds webview-side infrastructure for contributing new column renderers and implements an example for variables.

It also:
 - Replaces `Long`s with `BigInt`s (they can go in `Set`s)
 - Reorganizes the code more clearly into `entry-points`, `plugin`-side code, `webview`-side code, and `common` code that works in both plugin and webview contexts.

Open questions:
 - Rather than integrating the new webview-side services into the React lifecycle, I've made the module singletons. They could be managed differently.

#### How to test


1. Start a GDB debug session.
2. Open a memory view and view memory.
3. It should not display variables by default, and no request to retrieve variables should be made to the plugin side.
4. Open the advanced options section of the options widget and reveal the variables column.
5. The local variables available in the current frame should be retrieved, the regions they inhabit should be colored, and their names should appear in a third column:

![image](https://user-images.githubusercontent.com/62660806/229196418-d8ba98fa-2d54-49ed-abe0-a094873db370.png)

#### Review checklist


#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
